Feature: home page hero section created

Ahmed-Fahmy-Dev opened the pull request:
- JSX
- Styling

Ahmed-Fahmy-Dev committed:
feat: homepage hero section
- added image assets to public folder
- added json file with array of img urls

Ahmed-Fahmy-Dev committed:
feat: homepage hero section
- npm react material ui carousel package added

Ahmed-Fahmy-Dev committed:
feat: homepage hero section
- index module scss file deleted

Ahmed-Fahmy-Dev committed:
feat: homepage hero section
- fonts edited and pacifico fonts added

Ahmed-Fahmy-Dev committed:
feat: homepage hero section
- imageCarousel component created
- custom mui styled components created and used
- carousel styled

Ahmed-Fahmy-Dev committed:
feat: homepage hero section
- index.js file for components created for exporting all components

Ahmed-Fahmy-Dev committed:
feat: homepage hero section
- homepage hero section created
- custom material ui styled components created
- imageCarousel added with styled hero section headers

AltysS commented on client/src/components/ImageCarousel/ImageCarousel.js:
```diff
@@ -0,0 +1,30 @@
+    })();
+  }, []);
+
+  return (
```
The Carousel is not in the center of the page if I change the page scale. Please, see the screenshoot below.
But still, If I visit this page via the usual 100% screen scale everything seems to be OK. 
I`m not sure that this is a bug, but please double-check from your device as sometimes, my Chrome dev tools show me unexpected things.
![image](https://user-images.githubusercontent.com/112622259/198870144-403fb06a-4c88-426f-9dab-a27fca2bdcb1.png)

![image](https://user-images.githubusercontent.com/112622259/198870164-2255c8f8-7b9f-4717-871e-7a75a59b8bff.png)

Ahmed-Fahmy-Dev replied:
![image](https://user-images.githubusercontent.com/107735322/198877158-f7e10491-b4b1-4cb4-a324-7e0c0b864b65.png)


please check the different breakpoints view using chrome devtools responsive button. as changing the zoom scale is not an accurate way for this )

AltysS commented on client/src/pages/HomePage/HomePage.js:
```diff
@@ -1,5 +1,107 @@
+
+      <UkraineText>UKRAINE</UkraineText>
+
+      <Button sx={{ mt: '90px' }} endIcon={<ArrowRightAltIcon />}>
```
I guess that this button should navigate to catalog page

Ahmed-Fahmy-Dev replied:
the card task was to write the JSX and styling.
functionality is postponed later in future sprints

AltysS commented:
Have only two comments to review, the rest work seems perfect. @Ahmed-Fahmy-Dev , thank you!

Ahmed-Fahmy-Dev committed:
Merge branch 'develop' into feature/homePage-heroSection

HIMDanny commented on client/package.json:
```diff
@@ -17,6 +17,7 @@
 		"prop-types": "^15.8.1",
 		"react": "^18.2.0",
 		"react-dom": "^18.2.0",
+		"react-material-ui-carousel": "^3.4.2",
```
Maybe it is better to use Swiper instead, as we already using it in different component?

With your package we would have two sliders that do basically the same thing

HIMDanny replied:
I can explain the realization, if needed

Ahmed-Fahmy-Dev replied:
its not about the realization, we didnt agree on using one carousel library, so lets leave it this way now cause we are short of time on this sprint, and later on if needed and necessary we could use swiper.

on the other hand for educational purposes, i guess its ok to leave both libraries for future reference.

HIMDanny commented on client/public/index.html:
```diff
@@ -9,8 +9,10 @@
     <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
-    <link href="https://fonts.googleapis.com/css2?family=Montserrat:wght@100;300;400;500;600;700;800&display=swap"
+    <link
+      href="https://fonts.googleapis.com/css2?family=Montserrat:wght@100;300;400;500;600;700&family=Pacifico&display=swap"
```
As I noticed, you use only one font weight (400). I suggest deleting all unused font weights

Ahmed-Fahmy-Dev replied:
i will but on future fixes. I still didnt perform my final scan to the figma design and dont know what fonts we gonna need for our cart page and form design.

HIMDanny replied:
I don't really think that this font would be implemented in other places but hero section. It is more decorative. But ok, let postpone this question for future

HIMDanny commented on client/src/components/ImageCarousel/ImageCarousel.js:
```diff
@@ -0,0 +1,30 @@
+import React, { useEffect, useState } from 'react';
+import Carousel from 'react-material-ui-carousel';
```
Consider my previous comment about Swiper:
https://github.com/AltysS/travelAroundUkraine/pull/20#discussion_r1008883606

Ahmed-Fahmy-Dev replied:
Considered and already replied above.

HIMDanny commented on client/src/components/ImageCarousel/ImageCarousel.js:
```diff
@@ -0,0 +1,30 @@
+      indicators={false}
+    >
+      {images.map((url) => (
+        <img src={url} alt={url} key={url} />
```
Suggest changing a value of alt attribute to something more understandable, in case the photo wouldn't render. For example, "Ukraine landscape"

Ahmed-Fahmy-Dev replied:
Ack and done.

HIMDanny commented on client/src/components/index.js:
```diff
@@ -0,0 +1,4 @@
+import ImageCarousel from './ImageCarousel/ImageCarousel';
+import ErrorBoundary from './ErrorBoundary/ErrorBoundary';
+
+export { ImageCarousel, ErrorBoundary };
```
Don't you think that there is no necessity in adding ErrorBoundary to index? We use it only once and would not be using anywhere else

Ahmed-Fahmy-Dev replied:
yes used once but i guess from "saving the pattern" point of view, no exclusions to be made.

HIMDanny commented on client/src/index.module.scss:
```diff
@@ -1,11 +0,0 @@
-body {
-  margin: 0;
```
Please, add margin: 0 on body to global styles

Ahmed-Fahmy-Dev replied:
Done to App.module.scss as index.module.scss is not a needed file.

HIMDanny commented on client/src/pages/HomePage/HomePage.js:
```diff
@@ -1,5 +1,107 @@
 import React from 'react';
+import { Button, Container, Typography } from '@mui/material';
+import ArrowRightAltIcon from '@mui/icons-material/ArrowRightAlt';
+import { styled } from '@mui/material/styles';
```
You can combine "styled" with other components from "@mui/material"

Ahmed-Fahmy-Dev replied:
please check the below screenshot from the official documentation.

![image](https://user-images.githubusercontent.com/107735322/198905886-16c32d73-372d-49e5-a5b4-a6472cc4d8ec.png)

HIMDanny replied:
Yes, but you also can use `import { styled, ... } from '@mui/material'`, as you already using this path

HIMDanny commented on client/src/pages/HomePage/HomePage.js:
```diff
@@ -1,5 +1,107 @@
+
+  '.heroHeaderHighlighted': {
+    fontFamily: 'Pacifico',
+    fontWeight: 400,
```
Suggest deleting other unused font weights from this font
https://github.com/AltysS/travelAroundUkraine/pull/20#discussion_r1008886816

Ahmed-Fahmy-Dev replied:
Will do on future fixes when we are totally done with our design.

HIMDanny commented on client/src/pages/HomePage/HomePage.js:
```diff
@@ -1,5 +1,107 @@
+    <ImageCarousel />
+    <HeroContainer>
+      <TravelText>
+        TRAVEL <span className="heroHeaderHighlighted">AROUND</span>
```
I suggest adding `text-transform: uppercase` in styles instead of writing like this in html

Ahmed-Fahmy-Dev replied:
Ack. Done.

HIMDanny commented on client/src/pages/HomePage/HomePage.js:
```diff
@@ -1,5 +1,107 @@
+
+      <UkraineText>UKRAINE</UkraineText>
+
+      <Button sx={{ mt: '90px' }} endIcon={<ArrowRightAltIcon />}>
```
What about decreasing top margin so there would not be such a big space between text and button?

Ahmed-Fahmy-Dev replied:
Ack. Done. mt: 60px

HIMDanny commented on client/src/pages/HomePage/HomePage.js:
```diff
@@ -1,5 +1,107 @@
+      <UkraineText>UKRAINE</UkraineText>
+
+      <Button sx={{ mt: '90px' }} endIcon={<ArrowRightAltIcon />}>
+        TO OUR TOURS
```
What about using `text-transform: uppercase` here as well?

Ahmed-Fahmy-Dev replied:
Ack. Done.

HIMDanny commented on client/src/pages/HomePage/HomePage.module.scss:
```diff
@@ -0,0 +1,3 @@
+.heroSection {
+  position: relative;
```
**Controversial question:** if there would be no more styles in this file, maybe better to make styled component instead?

Ahmed-Fahmy-Dev replied:
i agree,  was waiting till the end of the first revision to roll back this style. Already Done.

HIMDanny commented on client/src/pages/HomePage/HomePage.js:
```diff
@@ -1,5 +1,107 @@
+  lineHeight: '64px',
+  color: 'rgba(255, 255, 255, 0)',
+  '-webkit-text-stroke-width': '2px',
+  '-webkit-text-stroke-color': 'rgb(41, 24, 194)',
```
Try using suggested properties:
![image](https://user-images.githubusercontent.com/58943798/198889949-e875ee9c-d474-401d-b203-71b73414ab69.png)

Ahmed-Fahmy-Dev replied:
Ack. Done ))

HIMDanny reviewed (commented):
Please, consider that on viewport 320px there is a problem with text positioning.
![image](https://user-images.githubusercontent.com/58943798/198889849-449f2ff8-d935-4060-952b-d8e1eba10b20.png)

Please, add `margin: 0` on body to global styles to properly show section.
![image](https://user-images.githubusercontent.com/58943798/198889896-7d838d26-8914-4ed2-8c94-47e58f45d670.png)

Ahmed-Fahmy-Dev commented:
- concerning text problem when viewport is 320 px    this is due to our container having a min width of 380px as per figma design.
- body margin 0 added.


![image](https://user-images.githubusercontent.com/107735322/198905348-4e4b320b-56dd-4dfe-a3de-7f6f51ce1448.png)

@HIMDanny

Ahmed-Fahmy-Dev committed:
feat: homepage hero section
- body margin 0 added in App module scss

Ahmed-Fahmy-Dev committed:
feat: homepage hero section
- image alt attribute edited to be more semantic

Ahmed-Fahmy-Dev committed:
feat: homepage hero section
- Box component replaced section tag
- text transform uppercase added
- webkit styling syntax edited to camel case instead of kebab case
- homepage  scss file deleted as styling was re written in Box sx prop

Ahmed-Fahmy-Dev committed:
feat: homepage hero section
- imported styled from @mui/material

Ahmed-Fahmy-Dev committed:
Merge branch 'develop' into feature/homePage-heroSection
